gossip/gossip.go

```diff
@@ -90,7 +90,7 @@ func (g *GossipMemberSet) Open() error {
 	g.memberlist, err = memberlist.Create(g.config.memberlistConfig)
 	g.mu.Unlock()
 	if err != nil {
-		return err
+		return fmt.Errorf("opening gossip member set: creating memberlist: %s", err)
```

Comment:
I've found that it's better to keep the context message local to the interior of the function. Whatever is calling this function should add the "opening gossip member set" bit, so this message would just be "creating memberlist:"